src/component-widget.js

```diff
@@ -72,6 +72,8 @@ export default class ComponentWidget {
         }
       }
 
+      // Clean up the old component
+      oldWidget.destroy()
```

Comment:
Nevermind, I missed the fact that it's the widget, not the component. Sorry for the noise.